api/controllers/Irc_EntryController.js

```diff
@@ -63,15 +64,17 @@ module.exports = {
   },
 
   process_event: function (request_body) {
-    if (request_body.operation === 'check in') {
-      return this.saveDetainee(request_body)
+    var eventName = request_body.operation;
+    var eventItem = eventsMap.find((mapItem) => typeof mapItem[eventName] !== 'undefined');
+
+    if (eventItem) {
+      const attrs = detaineeAttributes.create(request_body);
+      return this[eventItem[eventName] + 'Detainee'](attrs)
         .then(this.saveEvent.bind(this, request_body));
     }
     throw new ValidationError('Unknown');
```

Comment:
this should not be a `ValidationError` it's an `UnhandledError` with maybe `({operation:request_body.operation})` as args